src/uu/sysctl/src/sysctl.rs

```diff
@@ -0,0 +1,191 @@
+mod linux {
+    use std::path::{Path, PathBuf};
+    use uucore::error::UResult;
+    use walkdir::WalkDir;
+
+    const PROC_SYS_ROOT: &str = "/proc/sys";
+
+    pub fn get_all_sysctl_variables() -> Vec<String> {
+        let mut ret = vec![];
+        for entry in WalkDir::new(PROC_SYS_ROOT) {
+            match entry {
+                Ok(e) => {
+                    if e.file_type().is_file() {
+                        let var = e
+                            .path()
+                            .strip_prefix(PROC_SYS_ROOT)
+                            .expect("Always should be ancestor of of sysctl root");
+                        if let Some(s) = var.as_os_str().to_str() {
+                            ret.push(s.to_owned());
+                        }
+                    }
+                }
+                Err(e) => {
+                    uucore::show_error!("{}", e);
+                }
+            }
+        }
+        ret
+    }
+
+    pub fn normalize_var(var: &str) -> String {
+        var.replace('/', ".")
+    }
+
+    pub fn variable_path(var: &str) -> PathBuf {
+        Path::new(PROC_SYS_ROOT).join(var.replace('.', "/"))
+    }
+
+    pub fn get_sysctl(var: &str) -> UResult<String> {
+        Ok(std::fs::read_to_string(variable_path(var))?
+            .trim_end()
+            .to_string())
+    }
+
+    pub fn set_sysctl(var: &str, value: &str) -> UResult<()> {
+        std::fs::write(variable_path(var), value)?;
+        Ok(())
+    }
+}
+#[cfg(target_os = "linux")]
+use linux::*;
+
+#[cfg(target_os = "linux")]
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let vars = if matches.get_flag("all") {
+        get_all_sysctl_variables()
+    } else if let Some(vars) = matches.get_many::<String>("variables") {
+        vars.cloned().collect()
+    } else {
+        uu_app().print_help()?;
+        return Ok(());
+    };
+
+    for var_or_assignment in vars {
+        let mut splitted = var_or_assignment.splitn(2, '=');
+        let var = normalize_var(splitted.next().unwrap());
+
+        let value_to_print = if let Some(value_to_set) = splitted.next() {
+            if let Err(e) = set_sysctl(&var, value_to_set) {
+                if !matches.get_flag("ignore") {
+                    uucore::show_error!("error writing key '{}': {}", var, e);
+                    set_exit_code(1);
+                }
+                continue;
+            }
+            if matches.get_flag("quiet") {
+                continue;
+            }
+            value_to_set.to_string()
+        } else {
+            match get_sysctl(&var) {
+                Ok(v) => v,
+                Err(e) => {
+                    if !matches.get_flag("ignore") {
+                        uucore::show_error!("error reading key '{}': {}", var, e);
+                        set_exit_code(1);
+                    }
+                    continue;
+                }
+            }
```

Comment:
```suggestion
        get_sysctl(&var).unwrap_or_else(|e| {
    if !matches.get_flag("ignore") {
        uucore::show_error!("error reading key '{}': {}", var, e);
        set_exit_code(1);
    }
    return;
});
```

Reply:
I don't think I can do `continue` from inside the closure (to skip the print).. I guess splitting a function (that processes single var[=value]) ?

Reply:
Should be better now